src/Okta.Sdk/Internal/PayloadHandler.cs

```diff
@@ -0,0 +1,139 @@
+﻿// <copyright file="PayloadHandler.cs" company="Okta, Inc">
+// Copyright (c) 2020 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Http;
+
+namespace Okta.Sdk.Internal
+{
+    /// <summary>
+    /// A class responsible for providing body and message content.
+    /// </summary>
+    public abstract class PayloadHandler : IPayloadHandler
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PayloadHandler"/> class.
+        /// </summary>
+        public PayloadHandler()
+        {
+            Serializer = new DefaultSerializer();
+        }
+
+        static PayloadHandler()
+        {
+            _payloadHandlers.TryAdd("application/json", Default);
+        }
+
+        private static readonly ConcurrentDictionary<string, IPayloadHandler> _payloadHandlers = new ConcurrentDictionary<string, IPayloadHandler>();
+
+        private ISerializer Serializer { get; }
+
+        /// <inheritdoc/>
+        public string ContentType { get; protected set; }
+
+        /// <summary>
+        /// Gets or sets the content transfer encoding.
+        /// </summary>
+        protected string ContentTransferEncoding { get; set; }
+
+        private string _body;
+
+        /// <inheritdoc/>
+        public virtual string GetBody(HttpRequest httpRequest)
+        {
+            if (httpRequest.Payload != null && Serializer != null)
+            {
+                // don't re-serialize if already done
+                if (string.IsNullOrEmpty(_body))
+                {
+                    _body = Serializer.Serialize(httpRequest.Payload);
+                }
+            }
+
+            return _body;
+        }
+
+        /// <summary>
+        /// Get content for the specified request.
+        /// </summary>
+        /// <param name="httpRequest">The request whose content is returned.</param>
+        /// <returns>Content for the specified request.</returns>
+        protected abstract HttpContent GetContent(HttpRequest httpRequest);
```

Comment:
Nit: Let's rename this method to be more specific, something like: `GetRequestHttpContent`